src/etc/natvis/libcore.natvis

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<AutoVisualizer xmlns="http://schemas.microsoft.com/vstudio/debugger/natvis/2010">
+  <Type Name="core::ptr::Unique&lt;*&gt;">
+    <DisplayString>{{ Unique {*pointer.__0} }}</DisplayString>
+    <Expand>
+      <Item Name="[ptr]">pointer.__0</Item>
+    </Expand>
+  </Type>
+  <Type Name="core::ptr::Shared&lt;*&gt;">
+    <DisplayString>{{ Shared {*pointer.__0} }}</DisplayString>
+    <Expand>
+      <Item Name="[ptr]">pointer.__0</Item>
+    </Expand>
+  </Type>
+  <Type Name="core::option::Option&lt;*&gt;">
+    <DisplayString Condition="RUST$ENUM$DISR == 0x0">{{ None }}</DisplayString>
+    <DisplayString Condition="RUST$ENUM$DISR == 0x1">{{ Some {__0} }}</DisplayString>
+    <Expand>
+      <Item Name="[size]" ExcludeView="simple">(ULONG)(RUST$ENUM$DISR != 0)</Item>
+      <Item Name="[value]" ExcludeView="simple">__0</Item>
+      <ArrayItems>
+        <Size>(ULONG)(RUST$ENUM$DISR != 0)</Size>
+        <ValuePointer>&amp;__0</ValuePointer>
+      </ArrayItems>
+    </Expand>
+  </Type>
+  <Type Name="core::option::Option&lt;*&gt;" Priority="MediumLow">
```

Comment:
Setting the Priority here basically means "if you have problems with a higher priority visualization, come to me." The higher priority visualization relies on the RUST$ENUM$DISR member being present. If not, the code currently assumes that it's a cost-saving Enum where the NIL value is embedded within the type. This check is almost certainly wrong. Right now it's checking "are the first sizeof(PVOID) bytes of this structure 0"?